thirdparty/nginx/nginx-1.7.1-zimbra/src/core/ngx_zm_lookup.c

```diff
@@ -990,10 +992,14 @@ ngx_zm_lookup_create_request(ngx_zm_lookup_ctx_t *ctx)
         + sizeof ("X-Proxy-IP: ") - 1 + proxy_ip.len + sizeof(CRLF) - 1
         + sizeof ("Client-IP: ") - 1 + work->connection->addr_text.len + sizeof(CRLF) - 1;
 
-    if (work->isAdmin) {
+    if (work->type == admin) {
```

Comment:
Try to program this like
if (work->type == admin)
{
	//Do this
}
else if (work->type == zx)
{
	//Do that
}
This will reduce a extra check if work->type == admin

Reply:
dear @thefoolwhocodes, I think this optimization would have no impact because gcc & clang would optimize this condition.
In general having a jmp (of the else) is way more expensive than a simple cmp with a field (type) which is already in a register, so compilers would actually remove the 'else'.

Reply:
I feel it is nice to follow logical guidelines rather than leaving it for compiler optimization. I feel same is done for the parallel work that is done for java extensions..lets do it.
Thanks..

Reply:
Secondly writing it in this way makes me read it as following:
There can be scenario where the function takes both the argument and does some work. In such scenario i will use the above.

if (ConditionA)
    //do this
If (ConditionB)
    //do that